src/shared/cgroup-show.c

```diff
@@ -135,12 +135,12 @@ static int is_delegated(int cgfd, const char *path) {
         assert(cgfd >= 0 || path);
 
         r = getxattr_malloc(cgfd < 0 ? path : FORMAT_PROC_FD_PATH(cgfd), "trusted.delegate", &b);
-        if (r == -ENODATA) {
+        if (ERRNO_IS_XATTR_ABSENT(r)) {
```

Comment:
Please add `r < 0`, otherwise the condition may hit with the positive return value.